zerver/lib/rate_limiter.py

```diff
@@ -233,3 +243,21 @@ def incr_ratelimit(entity: RateLimitedObject) -> None:
                 count += 1
 
                 continue
+
+def rate_limit_entity(entity: RateLimitedObject) -> Tuple[bool, float]:
+    # Returns (ratelimited, secs_to_freedom)
+    ratelimited, time = is_ratelimited(entity)
+
+    if ratelimited:
+        statsd.incr("ratelimiter.limited.%s.%s" % (type(entity), str(entity)))
+
+    else:
+        try:
+            incr_ratelimit(entity)
+        except RateLimiterLockingException:
+            logger.warning("Deadlock trying to incr_ratelimit for %s:%s" % (
+                           type(entity), str(entity)))
```

Comment:
Can you modify a test to include `mock.patch` capture of this log line?  I'd like to see what output it creates for users to make sure it's correct, and we might as well have test coverage on the behavior.

Reply:
type(entity) ended up looking much uglier than I thought, so I changed that to type(entity).__name__ which will just print the classname now. And added a mock to capture that to test_hit_ratelimiterlockingexception in test_external.py in the rate_limit_entity commit